application/src/main/java/com/softserve/lv460/application/controller/LocalDeviceController.java

```diff
@@ -0,0 +1,52 @@
+package com.softserve.lv460.application.controller;
+
+import com.softserve.lv460.application.dto.localDevice.LocalDeviceRequest;
+import com.softserve.lv460.application.dto.location.LocationResponse;
+import com.softserve.lv460.application.entity.LocalDevice;
+import com.softserve.lv460.application.service.LocalDeviceService;
+import com.softserve.lv460.application.service.LocationService;
+import lombok.AllArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequestMapping("/location-device")
+@AllArgsConstructor
+public class LocalDeviceController {
+    private LocalDeviceService localDeviceService;
+    private LocationService locationService;
+
+    @PostMapping("/save")
+    public void save(@RequestBody LocalDeviceRequest localDeviceRequest) {
+        localDeviceService.save(localDeviceRequest);
+    }
+
+    @GetMapping("/find-all")
+    public List<LocalDevice> findAll() {
+        return localDeviceService.findAll();
+    }
+
+    @GetMapping("/find-all-in-location/{location_id}")
+    public ArrayList<LocalDevice> findByLocation(@PathVariable("location_id") Long id){
+        return localDeviceService.findAllByLocation(locationService.findOne(id));
+    }
+
+    @GetMapping("find/{uuid}")
+    public LocalDevice findOne(@PathVariable("uuid") String uuid) throws Exception {
+        return localDeviceService.findByUuid(uuid);
+    }
+
+    @PutMapping("/update/{uuid}/{location_id}")
```

Comment:
What this service method does? updates device's location? what if we need more in future? PUT should accept the entire object you want to update